src/libraries/Common/tests/System/Net/RandomReadWriteSizeStream.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Test.Common
+{
+    // Wrapper stream to artificially chop reads and writes to smaller chunks.
+    public class RandomReadWriteSizeStream : Stream
+    {
+        private readonly Stream _innerStream;
+        private readonly int _maxSize;
+
+        public RandomReadWriteSizeStream(Stream stream, int maximumSize = int.MaxValue)
```

Comment:
A couple nits:

I think we should have a more descriptive name instead of `maximumSize`. I'd suggest `maxChunkSize`. In general we usually don't spell out `maximum`, we just use `max` here.

And we should be consistent here between the param name  and the field name.

Reply:
updated